src/process_viewer.rs

```diff
@@ -60,6 +64,77 @@ fn update_window(list: &gtk::ListStore, system: &Rc<RefCell<sysinfo::System>>,
     }
 }
 
+fn parse_quote(line: &str, quote: char) -> Vec<CString> {
+    let args = line.split(quote).collect::<Vec<&str>>();
+    let mut out_args = vec!();
+
+    for (num, arg) in args.iter().enumerate() {
+        if num != 1 {
+            out_args.extend_from_slice(&parse_entry(arg));
+        } else {
+            out_args.push(CString::new(*arg).unwrap());
+        }
+    }
+    out_args
+}
+
+// super simple parsing
+fn parse_entry(line: &str) -> Vec<CString> {
+    match (line.find('\''), line.find('"')) {
+        (Some(x), Some(y)) => {
+            if x < y {
+                parse_quote(line, '\'')
+            } else {
+                parse_quote(line, '"')
+            }
+        }
+        (Some(_), None) => parse_quote(line, '\''),
+        (None, Some(_)) => parse_quote(line, '"'),
+        (None, None) => line.split(' ').into_iter()
+                                       .map(|s| CString::new(s).unwrap())
+                                       .collect::<Vec<CString>>(),
+    }
+}
+
+fn start_detached_process(line: &str) -> Option<i32> {
+    let args = parse_entry(line);
+    let command = args[0].clone();
+
+    unsafe {
+        let pid = libc::fork();
+
+        if pid < 0 {
+            libc::exit(2); // error
+        } else if pid == 0 {
+            libc::umask(0);
+            if libc::setsid() < 0 {
+                libc::exit(3);
+            }
+            libc::chdir("/".as_ptr() as *const i8);
+            let stdin = libc::fdopen(0, "r".as_ptr() as *const i8);
+            let stdout = libc::fdopen(1, "w".as_ptr() as *const i8);
+            let stderr = libc::fdopen(2, "w".as_ptr() as *const i8);
+            libc::freopen("/dev/null".as_ptr() as *const i8, "r".as_ptr() as *const i8, stdin);
```

Comment:
Yes, I forgot to upload the updated version. But the error is even before calling `execv`:

```Shell
[xcb] Unknown sequence number while processing queue
[xcb] Most likely this is a multi-threaded client and XInitThreads has not been called
[xcb] Aborting, sorry about that.
process_viewer: ../../src/xcb_io.c:274: poll_for_event: Assertion `!xcb_xlib_threads_sequence_lost' failed.
```

Since it's supposed to be a detached process, I'm quite surprised of this output.